lib/terraforming/resource/security_group.rb

```diff
@@ -101,6 +101,54 @@ def permission_attributes_of(security_group, permission, type)
         attributes
       end
 
+      def dedup_permissions(security_group)
+        grouped_ingress = security_group.ip_permissions.group_by {|perm| [perm.ip_protocol, perm.to_port, perm.from_port]}
+        grouped_egress = security_group.ip_permissions_egress.group_by {|perm| [perm.ip_protocol, perm.to_port, perm.from_port]}
+
+        security_group.ip_permissions = []
+        security_group.ip_permissions_egress = []
+
+        grouped_ingress.each do |range, perms|
+          if perms.length == 1
+            security_group.ip_permissions << perms.first
+            grouped_ingress.delete(range)
```

Comment:
Is this nesseary to delete element?
It seems that this sentence makes no sense, because this sentence deletes an element at current position and `grouped_ingress` is never used after here.
It works well without this sentence with my environment.

In either case, I think deleting elements in `each` block is not a good way...

(other `delete` sentences are the same)